src/NServiceBus.MessagingBridge/Configuration/HostBuilderExtensions.cs

```diff
@@ -33,22 +34,23 @@ public static IHostBuilder UseNServiceBusBridge(this IHostBuilder hostBuilder, A
             var deferredLoggerFactory = new DeferredLoggerFactory();
             LogManager.UseFactory(deferredLoggerFactory);
 
-            hostBuilder.ConfigureServices((hostBuilderContext, serviceCollection) =>
+            _ = hostBuilder.ConfigureServices((hostBuilderContext, serviceCollection) =>
             {
                 var bridgeConfiguration = new BridgeConfiguration();
 
                 bridgeConfigurationAction(hostBuilderContext, bridgeConfiguration);
 
-                serviceCollection.AddSingleton(sp => bridgeConfiguration.FinalizeConfiguration(sp.GetRequiredService<ILogger<BridgeConfiguration>>()));
-
-                serviceCollection.AddSingleton(deferredLoggerFactory);
-                serviceCollection.AddSingleton<IHostedService, BridgeHostedService>();
-                serviceCollection.AddSingleton<IStartableBridge, StartableBridge>();
-                serviceCollection.AddSingleton<EndpointProxyFactory>();
-                serviceCollection.AddSingleton<SubscriptionManager>();
-                serviceCollection.AddSingleton<EndpointRegistry>();
-                serviceCollection.AddSingleton<IEndpointRegistry>(sp => sp.GetRequiredService<EndpointRegistry>());
-                serviceCollection.AddSingleton<IMessageShovel, MessageShovel>();
+                _ = serviceCollection.AddSingleton(sp => bridgeConfiguration.FinalizeConfiguration(sp.GetRequiredService<ILogger<BridgeConfiguration>>()));
```

Comment:
this seems like something and IDE did, and is not likely what we want.

Reply:
It's UGLY. the other options is removing all the repeated serviceCollection. and chain the AddSignleton calls

Reply:
I apologize, that was rude, what I meant to say is I haven't seen discards used in this context in applications before, and it gives a somewhat jarring impression.